PSKoans/Koans/Constructs and Patterns/AboutXml.Koans.ps1

```diff
@@ -0,0 +1,740 @@
+    <drive fileSystem="NTFS">
+        <letter>C:</letter>
+    </drive>
+    <drive fileSystem="NTFS">
+        <letter>D:</letter>
+    </drive>
+    <drive fileSystem="FAT32">
+        <letter>G:</letter>
+    </drive>
+</drives>
+'@
+
+            # The @ character is added in front of a label when testing the value of an attribute.
+            $elements = $xml.SelectNodes('/drives/drive[@fileSystem="NTFS"]/letter')
+
+            @('____', '____') | Should -Be $elements.'#text'
+        }
+
+        It 'always returns a list when using SelectNodes' {
+            $xml = [Xml]@'
+<drives>
+    <drive fileSystem="NTFS">
+        <letter>C:</letter>
+    </drive>
+    <drive fileSystem="FAT32">
+        <letter>G:</letter>
+    </drive>
+</drives>
+'@
+
+            $element = $xml.SelectNodes('/drives/drive[@fileSystem="FAT32"]/letter')
+
+            $ExpectedType = [____]
+
+            $element.GetType() | Should -Be $ExpectedType
+        }
+    }
+
+    Context 'Writing XML documents in PowerShell' {
+        # .NET offers the XmlWriter types to create XML content from scratch.
+
+        It 'can create an XML file' {
+            $path = Join-Path -Path $TestDrive -ChildPath 'document.xml'
+            $xmlWriter = [System.Xml.XmlWriter]::Create($path)
+
+            # A root element can be added to the document
+            $xmlWriter.WriteStartElement('drives')
+
+            <#
+                When the XML document is complete, the buffer can be flushed, ensuring all content has been written
+                to the stream. Then the document can be closed, allowing other processes to use the file.
+            #>
+            $xmlWriter.Flush()
+            $xmlWriter.Close()
+
+            $ExpectedContent = '____'
+
+            $path | Should -FileContentMatch $ExpectedContent
+        }
+
+        It 'has configurable settings' {
+            <#
+                By default the XmlWriter will create an XML declaration and all content will be written to a single
+                line.
+
+                XmlWriterSettings can be provided to change this and other behaviours. Settings cannot be modified
+                after the XmlWriter has been created.
+            #>
+
+            $path = Join-Path -Path $TestDrive -ChildPath 'document.xml'
+
+            # If Indent is et to $true, line breaks will be added. By default two-spaces are used to indent content.
+            $settings = [System.Xml.XmlWriterSettings]@{
+                OmitXmlDeclaration = $true
+                Indent             = $true
+            }
+            $xmlWriter = [System.Xml.XmlWriter]::Create($path, $settings)
+
+            $xmlWriter.WriteStartElement('drives')
+            $xmlWriter.WriteStartElement('drive')
+
+            # Closing the document will automatically add missing end elements.
+            $xmlWriter.Flush()
+            $xmlWriter.Close()
+
+            $content = Get-Content $path
+
+            __ | Should -Be $content.Count
```

Comment:
Is it worth adding a here-string with blanks that users can fill in, so they get a more complete visual of how a file written in this way will end up looking?

Reply:
Alternately you can do line-by-line assertions with blanks via indexes of `$content`. 🙂